### PR TITLE
Add consistent examples to README

### PR DESCRIPTION
**Pull request by dfed**
This PR gets each section of the README re-visiting the same ordering test to make explicit how these types solve the problem statement in the `Task Ordering and Swift Concurrency` section of the README.

There's still room to improve the explanation of these types, but we're making progress here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,13 +14,21 @@ Tasks sent from a synchronous context to an asynchronous context in Swift Concur
 
 ```swift
 @MainActor
-func test_mainActor_taskOrdering() async {
-    var counter = 0
+func testMainActorTaskOrdering() async {
+    actor Counter {
```
Comment (dfed) [inline README.md:18]: We don't _need_ a `Counter` type in this example but the below examples do need it. For the sake of consistency with the below additions I made this example test a dash more complicated.

File: README.md [hunk 2]
```diff
@@ -51,14 +59,35 @@ queue.async {
     This task begins execution once the above one-second sleep completes.
     */
 }
-Task {
```
Comment (dfed) [inline README.md:54]: This `Task` was meant to show that we were entering an async context, but I'm realizing now that the inclusion of the `Task` is more confusing than not. Nothing within the `Task` closure changed – I just removed the `Task` enclosure. I did the same thing further down as well.

File: README.md [hunk 3]
```diff
@@ -80,14 +109,35 @@ queue.async {
...
+    let queue = ActorQueue()
+    for iteration in 1...100 {
+        queue.async {
+            let incrementedCount = await counter.increment()
+            XCTAssertEqual(incrementedCount, iteration) // always succeeds
+        }
     }
+    await queue.await { }
 }
 ```
```
Comment (dfed) [inline README.md:142]: Ideally we'd explain better how the `ActorQueue` and `FIFOQueue` differ in the above example test, but I'm wary of trying to communicate too many ideas at once.

File: README.md (hunk 2 above)
Comment (bachand) [reply README.md:54]: This makes sense to me. The reader can infer from the `await` below that we're in an asynchronous context.

File: README.md [hunk 4]
```diff
@@ -51,14 +59,35 @@ queue.async {
...
+
+    let counter = Counter()
+    let queue = FIFOQueue()
+    for iteration in 1...100 {
+        queue.async {
+            let incrementedCount = await counter.increment()
+            XCTAssertEqual(incrementedCount, iteration) // always succeeds
+        }
     }
+    await queue.await { }
```
Comment (bachand) [inline README.md:90]: Maybe clarify with a comment that we're ensuring all tasks complete before we end the test?

File: README.md [hunk 5]
```diff
@@ -80,14 +109,35 @@ queue.async {
...
+
+    let counter = Counter()
+    let queue = ActorQueue()
+    for iteration in 1...100 {
+        queue.async {
+            let incrementedCount = await counter.increment()
+            XCTAssertEqual(incrementedCount, iteration) // always succeeds
+        }
     }
+    await queue.await { }
```
Comment (bachand) [inline README.md:140]: Maybe we also clarify the purpose of this line.

**Review by bachand — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (dfed) [reply README.md:90]: I'll fix this in #9 which builds on top of this change.

File: README.md (hunk 5 above)
Comment (dfed) [reply README.md:140]: I'll fix this in #9 which builds on top of this change.

File: README.md (hunk 4 above)
Comment (dfed) [reply README.md:90]: c4d455f

**Pull request merged**